x-pack/plugins/infra/public/alerting/log_threshold/components/expression_editor/editor.tsx

```diff
@@ -255,6 +255,13 @@ export const Editor: React.FC<
     setHasSetDefaults(true);
   });
 
+  const shouldShowGroupByOptimizationWarning = useMemo(() => {
+    const hasSetGroupBy = alertParams.groupBy && alertParams.groupBy.length > 0;
+    const hasSetOptimizableThresholdComparator =
+      alertParams.count && alertParams.count.comparator === Comparator.GT;
```

Comment:
Can we share this predicate implementation with the executor to keep them in sync?